src/features/bivariate_color_manager/components/Sublist/Sublist.tsx

```diff
@@ -0,0 +1,100 @@
+import clsx from 'clsx';
+import { sortByKey } from '~utils/common';
+import {
+  CSSTransitionWrapper,
+  fadeClassNames,
+} from '../CssTransitionWrapper/CssTransitionWrapper';
+import s from './Sublist.module.css';
+import type {
+  BivariateColorManagerData,
+  BivariateColorManagerDataValue,
+  TableDataValue,
+} from '~features/bivariate_color_manager/atoms/bivariateColorManagerResource';
+import type {
+  BivariateColorManagerAtomState,
+  LayerSelectionInput,
+} from '~features/bivariate_color_manager/atoms/bivariateColorManager';
+
+type SentimentsCombinationsListProps = {
+  data: BivariateColorManagerData;
+  setLayersSelection: (input: LayerSelectionInput) => void;
+  layersSelection: BivariateColorManagerAtomState['layersSelection'];
+};
+
+const sortDescendingByQuality = sortByKey<TableDataValue>('quality', 'desc');
+
+type SublistProps = {
+  open: boolean;
+  rowData: BivariateColorManagerDataValue;
+  rowKey: string;
+  setLayersSelection: SentimentsCombinationsListProps['setLayersSelection'];
+  layersSelection: SentimentsCombinationsListProps['layersSelection'];
+};
+
+type Plane = 'vertical' | 'horizontal';
+
+const Sublist = ({
+  open = false,
+  rowData,
+  rowKey: key,
+  setLayersSelection,
+  layersSelection,
+}: SublistProps) => {
+  const { vertical, horizontal } = rowData;
+  const verticalList = Object.values(vertical).sort(sortDescendingByQuality);
+  const horizontalList = Object.values(horizontal).sort(
+    sortDescendingByQuality,
+  );
+  const selectlayer = (
+    plane: Plane,
+    quotientIndicator: TableDataValue,
+  ): void => {
+    setLayersSelection({
+      key,
+      [plane]: quotientIndicator,
+    });
+  };
+
+  const isLayerSelected = (plane: Plane, quotientIndicator: TableDataValue) =>
+    layersSelection &&
+    quotientIndicator.label === layersSelection?.[plane]?.label;
+
+  // eslint-disable-next-line react/display-name
+  const renderItem = (plane: Plane) => (quotientIndicator: TableDataValue) =>
+    (
+      <div
+        onClick={() => selectlayer(plane, quotientIndicator)}
+        className={clsx(
+          s.sublistRow,
+          isLayerSelected(plane, quotientIndicator) && s.sublistRowSelected,
+        )}
+        key={quotientIndicator.label}
+      >
+        {quotientIndicator.label}
+      </div>
+    );
+
+  const renderVerticalItem = renderItem('vertical');
+  const renderHorizontalItem = renderItem('horizontal');
+
+  return (
+    <CSSTransitionWrapper
+      in={open}
+      timeout={300}
+      unmountOnExit
+      appear
+      classNames={fadeClassNames}
+    >
+      {(ref) => (
+        <tr ref={ref} className={s.sublist}>
+          <td />
+          <td />
+          <td>{verticalList.map(renderVerticalItem)}</td>
+          <td>{horizontalList.map(renderHorizontalItem)}</td>
+        </tr>
```

Comment:
why do we need table tags, especially empty ones?

Reply:
we have 4 columns, and we need to locate sublists on last 2 columns, first 2 should remain empty.
<img width="929" alt="Screenshot 2022-07-28 at 15 06 15" src="https://user-images.githubusercontent.com/20676525/181501028-83874732-c8ff-4a67-b8a1-4d1d1c29ae19.png">

maybe there is more clear way of how to skip first 2 columns idk
anyway feel free to share of how to make it clearer :)

Reply:
I see, proboply it's the easiest way. However I'd call Sublist component with a less abstract name as it's not a shared component - like what it does\shows us? Availible layers? Featured layers?

Reply:
if to think from entities side it's a list of sentiments combinations
so maybe SentimentsCombinationsList?

Reply:
oh, god, our root table component has this name  🤔